src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java

```diff
@@ -2084,12 +2103,95 @@ public EntryResult tMedicationDispense2MedicationDispense(
 	}
 
 	@Override
-	public Bundle tMedicationInformation2Medication(MedicationInformation cdaMedicationInformation) {
-		/*
-		 * Since MedicationInformation is a ManufacturedProduct instance with a specific
-		 * templateId, tManufacturedProduct2Medication should satisfy the required
-		 * mapping for MedicationInformation
-		 */
+	public EntryResult medicationSupplyOrder2MedicationRequest(MedicationSupplyOrder cdaSupplyOrder,
+			IBundleInfo localBundleInfo) {
+		EntryResult result = new EntryResult();
+
+		if (cdaSupplyOrder == null || cdaSupplyOrder.isSetNullFlavor())
+			return result;
+
+		MedicationRequest medRequest = new MedicationRequest();
+		result.addResource(medRequest);
+
+		// patient
+		medRequest.setSubject(getPatientRef());
+
+		// resource id
+		IdType resourceId = new IdType("MedicationRequest", getUniqueId());
+		medRequest.setId(resourceId);
+
+		// meta.profile
+		if (Config.isGenerateDafProfileMetadata())
+			medRequest.getMeta().addProfile(Constants.PROFILE_DAF_MEDICATION_REQUEST);
+
+		// id -> identifier
+		if (cdaSupplyOrder.getIds() != null && !cdaSupplyOrder.getIds().isEmpty()) {
+			for (II ii : cdaSupplyOrder.getIds()) {
+				if (ii != null && !ii.isSetNullFlavor()) {
+					medRequest.addIdentifier(dtt.tII2Identifier(ii));
+				}
+			}
+		}
+
+		// effectiveTime.low -> authoredOn
+		if (cdaSupplyOrder.getEffectiveTimes() != null && !cdaSupplyOrder.getEffectiveTimes().isEmpty()) {
```

Comment:
per the spec looks like effectiveTime high will come in (not low).

Reply:
Good point I'll change that.

Reply:
On second thought High is required but I don't know if it would map to authoredOn?

Reply:
how about validity period?

Reply:
yeah map to that.